stringfuzz/generators/regex.py

```diff
@@ -70,24 +100,32 @@ def make_regex_string(min_length, max_length):
 
     return smt_str_to_re(smt_str_lit(string))
 
-def make_random_term(depth):
+def make_random_term(depth, operator=None):
     if depth == 0:
         return make_regex_string(_literal_min, _literal_max)
 
-    subterm = make_random_term(depth - 1)
+    if _operator_type == OPERATOR_ALTERNATING:
+        if operator == None:
+            operator = _operator_list[0]
+    else:
+        operator = random.choice(_operator_list)
```

Comment:
Instead of calculating the operator for *this* level, what do you think of calculating the operator for the *next* level here? That way the default argument won't be necessary (it can be passed initially in `make_regex`, just like `depth` is), the "ignored" comment won't be necessary below, and also the "next operator" calculation of `OPERATOR_ALTERNATING` will happen right in this if-statement.

Reply:
A slight note: I passed the initial operator index in ``make_random_terms`` instead.